waiter/src/waiter/auth/spnego.clj

```diff
@@ -69,17 +69,37 @@
       (cookies/cookies-response)))
 
 (defn response-503-temporarily-unavailable
-  "Tell the client you're overloaded and would like them to try later"
+  "Tell the client you're overloaded and would like them to try later."
   []
-  (log/info "triggering 401 negotiate for spnego authentication")
+  (log/info "triggering 503 temporarily unavailable")
   (counters/inc! (metrics/waiter-counter "core" "response-status" "503"))
   (meters/mark! (metrics/waiter-meter "core" "response-status-rate" "503"))
-  (-> (rr/response "Too many Kerberos authentication requests")
+  (-> (rr/response "Too many Kerberos authentication requests; consider enabling cookies in your client")
       (rr/status 503)
       (rr/header "content-type" "text/plain")
       (utils/attach-waiter-source)
       (cookies/cookies-response)))
 
+(defn response-401-negotiate-or-503-temporarily-unavailable
+  "Depending upon the rate of requests requiring kerberos auth,
+   either tell the client you're overloaded and would like them to try later,
+   or tell the client you'd like them to use kerberos to authenticate."
+  [max-one-minute-rate-per-host request]
+  (let [host-name (-> request
+                    :headers
+                    (get "host")
+                    (or "_unknown_")
```

Comment:
`_no_host_`?  Could we better indicate that the client didn't specify the host rather then that we don't know the host?

Reply:
Do we intend to limit all on-the-fly requests to Waiter's hostname to the same rate as an individual service?